opensearchpy/connection/http_requests.py

```diff
@@ -187,13 +191,15 @@ def perform_request(  # type: ignore
             "allow_redirects": allow_redirects,
         }
         send_kwargs.update(settings)
+        start = time.time()
         try:
             response = self.session.send(prepared_request, **send_kwargs)
             duration = time.time() - start
             raw_data = response.content.decode("utf-8", "surrogatepass")
         except reraise_exceptions:
             raise
         except Exception as e:
+            duration = time.time() - start
```

Comment:
I don't see this variable being used anywhere since `ConnectionError` is raised after.